src/mono/mono/mini/mini-amd64.c

```diff
@@ -1023,8 +1023,7 @@ get_call_info (MonoMemPool *mp, MonoMethodSignature *sig)
 
 			if ((klass == swift_self || klass == swift_indirect_result) && sig->pinvoke) {
 				guint32 size = mini_type_stack_size_full (m_class_get_byval_arg (klass), NULL, sig->pinvoke && !sig->marshalling_disabled);
-				g_assert (size == 8);
-
+				g_assert (size == SIZEOF_VOID_P);
```

Comment:
```suggestion
				g_assert (size == TARGET_SIZEOF_VOID_P);
```

I think for AOT we want the target arch, not the host